doc/doc_comments.mld

```diff
@@ -0,0 +1,42 @@
+{0 Doc-comments language reference}
+
+OCamlFormat uses {{:https://github.com/ocaml/odoc}odoc} to parse doc-comments (also referred to as doc-strings), and hence it inherits the accepted language from [odoc] (detailed in {{:https://ocaml.github.io/odoc/odoc_for_authors.html}odoc's documentation}).
+
+Here is an example showing a few useful elements:
+
+{%html:
+<pre>
+<code class="ml hljs ocaml">(** {0 Adding integers} *)
+
+(** {1 Exception} *)
+
+(** Raised in case of integer overflow *)
+exception Int_overflow
+
+(** {1 Function definition} *)
+
+(** [add ~x ~y] returns [x + y] or raises an exception in case of integer overflow.
+    Usage:
+    {@ocaml some_metadata[
+    # add ~x:1 ~y:2 ;;
+    - : int = 3
+    ]}
+
+    Here is a basic diagram:
+    {v
+          add  ~x:1   ~y:2
+                  \   /
+                   (+)
+                    |
+                    3
+    v}
+
+    Notes:
+    - {_ check} that exception {!exception:Int_overflow} is {b not raised};
+    - have a look at {!module:Int} ({{:https://ocaml.org/api/Int.html}Documentation}).
```

Comment:
The link to the documentation is weird, the reference is already a better link to the same documentation because it stays on the same site. Why not a link to the RWO chapter about the encoding of ints: https://dev.realworldocaml.org/runtime-memory-layout.html#scrollNav-2-1

Reply:
The idea was to show the syntax for links, but yeah it doesn't have much and it's not very clear.